packages/@aws-cdk/aws-codedeploy/lib/deployment-configuration.ts

```diff
@@ -0,0 +1,155 @@
+    public abstract readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+
+    public export(): CustomServerDeploymentConfigurationRefProps {
+        return {
+            deploymentConfigurationName: new cdk.Output(this, 'DeploymentConfigurationName', {
+                value: this.deploymentConfigurationName,
+            }).makeImportValue(),
+        };
+    }
+}
+
+class ImportedCustomServerDeploymentConfigurationRef extends CustomServerDeploymentConfigurationRef {
+    public readonly deploymentConfigurationName: DeploymentConfigurationName;
+    public readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+
+    constructor(parent: cdk.Construct, id: string, props: CustomServerDeploymentConfigurationRefProps) {
+        super(parent, id);
+
+        this.deploymentConfigurationName = props.deploymentConfigurationName;
+        this.deploymentConfigurationArn = deploymentConfigurationName2Arn(this.deploymentConfigurationName);
+    }
+}
+
+/**
+ * The possible representations of a healthy-host value -
+ * either percent-based, or an absolute number.
+ */
+export enum HostsRepresentation {
+    NUMBER = 'HOST_COUNT',
+    PERCENTAGE = 'FLEET_PERCENT'
+}
+
+export interface MinimumHealthyHosts {
+    readonly representation: HostsRepresentation;
+
+    readonly value: number;
+}
+
+/**
+ * Construction properties of {@link CustomServerDeploymentConfiguration}.
+ */
+export interface CustomServerDeploymentConfigurationProps {
+    /**
+     * The physical, human-readable name of the Deployment Configuration.
+     *
+     * @default a name will be auto-generated
+     */
+    deploymentConfigurationName?: string;
+
+    /**
+     * The minimum healhty hosts setting for this EC2/on-premise Deployment Configuration.
+     */
+    minimumHealthyHosts: MinimumHealthyHosts;
+}
+
+/**
+ * A custom Deployment Configuration for an EC2/on-premise Deployment Group.
+ */
+export class CustomServerDeploymentConfiguration extends CustomServerDeploymentConfigurationRef {
+    public readonly deploymentConfigurationName: DeploymentConfigurationName;
+    public readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+    public readonly minimumHealthyHosts: MinimumHealthyHosts;
+
+    constructor(parent: cdk.Construct, id: string, props: CustomServerDeploymentConfigurationProps) {
+        super(parent, id);
+
+        this.minimumHealthyHosts = props.minimumHealthyHosts;
+
+        const resource = new cloudformation.DeploymentConfigResource(this, 'Resource', {
+            deploymentConfigName: props.deploymentConfigurationName,
+            minimumHealthyHosts: {
+                type: this.minimumHealthyHosts.representation.toString(),
+                value: this.minimumHealthyHosts.value,
+            },
+        });
+
+        this.deploymentConfigurationName = resource.ref;
+        this.deploymentConfigurationArn = deploymentConfigurationName2Arn(this.deploymentConfigurationName);
```

Comment:
Why not just implement this as a calculated property in the `DeploymentConfigRef`? (seems like this is repeating everywhere)

Reply:
Because you need to call `super` in the constructor as the first statement, and the name is a property of the constructed `DeploymentConfigResource`, so there is no way to pass it in the `super` call (well, technically we could do it, but it would be ugly).